docs/architecture/009-review-and-form-model.md

```diff
@@ -0,0 +1,93 @@
+   Model common fields as a form that can be included in the workflow.
+   Application contains form fields from all forms of all catalogue items and the forms of one workflow.
+   Review request is sent with certain organizations selected and it controls which fields are shown to the reviewer.
+   Application contains forms and fields from catalogue items and workflows.
+
+## Discussion
+
+As this is deemed most important feature to improve, we can disregard **option 1** doing nothing.
+
+Modeling fields into catalogue items does not seem like a sensible direction, it would introduce exceptions. We should therefore disregard **option 2** and prefer the others e.g. **option 3** modeling the fields into a form.
+
+Current application bundling logic bundles all catalogue items to a single application if they share the form and workflow. It seems like a natural extension of the model to allow several forms to be included and combined.
+
+A naive version of form merging just duplicates all the fields in the form and a smart version merges similar fields so the applicant has to answer only once each. However we don't separately model each field so it's difficult to identify which field is exactly the same and can be merged. One would have to make sure each field has a globally distinctive id. We know that there would be around 1000 potential forms that would have to be made with shared and separate parts. Maintaining the forms and their shared parts would be too much work and would need an extension of the editor as well.
+
+A possible variant of form merging is also forcing shared parts to be in a separate form. Then there are a lot of forms but fields are not duplicated. However the applicant must somehow always choose wisely and include the shared catalogue item. This can be fixed with *option 6* as shown later.
+
+All in all while multiple forms, and **option 3** are a sensible extension to our model, it certainly requires large changes to the logic all around the codebase.
+
+In the **option 4** we would make a new repeating functionality that would repeat per applied catalogue item. Modeling each reviewer organization as a catalogue item seems like a business decision, not something for us. It requires the organization to either be able to statically know each option, or have a process for adding missing ones. While the repeating logic is sound, we can also consider **option 5** as the same, but not having the need to couple this into catalogue items. So lets prefer **option 5** and not this one.
+
+A field type where you can "press add button" to add more rows sounds also like a natural extension to the types of fields we have. Also repeating fields based on the count of a previous answer sounds sensible. Some kind of grouping logic may be needed to repeat fields in a group (ABCABCABC and not AAABBBCCC). Therefore **option 5** seems like a sensible extension that we should consider.
```

Comment:
I'm not sure what these repeated fields are. Perhaps they should be covered in Background?

Reply:
They don't exist so they are not background. It is a concept that answer is repeated for some reason, not just one per field.

Reply:
Now with added wording and explanation before option list.